src/upgrades/pre_0_10_0/cas_migration.rs

```diff
@@ -77,192 +81,79 @@ impl CasMigration {
-                        return Err(PrepareUpgradeError::custom(
-                        format!("New info.json does not match old info.json when upgrading CA '{}'. Please downgrade to the previous version and provide a bug report to rpki-team@nlnetlabs.nl.", handle),
-                    ));
-                    }
+        let new_command = match old_effect {
+            UnconvertedEffect::Error { msg } => new_command_builder.finish_with_error(msg),
+            UnconvertedEffect::Success { events } => {
+                let mut full_events: Vec<CertAuthEvent> = vec![]; // We just had numbers, we need to include the full events
+                for old_event in events {
+                    full_events.push(old_event.try_into()?);
                 }
+                new_command_builder.finish_with_events(full_events)
             }
+        };
 
-            // Verify migration
-            info!("Will verify the migration by rebuilding CA '{}' events", &scope);
-            let ca = self.new_agg_store.get_latest(&handle).map_err(|e| {
-                PrepareUpgradeError::Custom(format!(
-                    "Could not rebuild state after migrating CA '{}'! Error was: {}.",
-                    handle, e
-                ))
-            })?;
-
-            // Store snapshot to avoid having to re-process the deltas again in future
-            self.new_agg_store.store_snapshot(&handle, ca.as_ref()).map_err(|e| {
-                PrepareUpgradeError::Custom(format!(
-                    "Could not save snapshot for CA '{}' after migration! Disk full?!? Error was: {}.",
-                    handle, e
-                ))
-            })?;
-
-            // Load the CaObjects for this CA and convert it.
-            info!("Will migrate the current repository objects for CA '{}'", handle);
-            self.ca_objects_migration.prepare_new_data_for(&handle)?;
-
-            info!("Verified migration of CA '{}'", handle);
-        }
-
-        match mode {
-            UpgradeMode::PrepareOnly => {
-                info!(
-                    "Prepared migrating CAs to Krill version {}. Will save progress for final upgrade when Krill restarts.",
-                    KRILL_VERSION
-                );
-            }
-            UpgradeMode::PrepareToFinalise => {
-                self.clean_migration_help_files()?;
-                info!("Prepared migrating CAs to Krill version {}.", KRILL_VERSION);
-            }
-        }
-
-        Ok(())
-    }
-
-    fn deployed_store(&self) -> &KeyValueStore {
-        &self.current_kv_store
+        Ok(Some(new_command))
     }
 
-    fn preparation_store(&self) -> &KeyValueStore {
-        &self.new_kv_store
+    /// Override post migration, we need to do extra stuff.
+    fn post_migration(&self, handle: &MyHandle) -> UpgradeResult<()> {
+        info!("Will migrate the current repository objects for CA '{}'", handle);
```

Comment:
How can "will migrate" which seems to refer to a future migration be correct in a *post* migration step? Should this say "Will verify" instead of "Will migrate"? And should it refer to "CA" here or be more generic?

Reply:
It's part of the initial migration but post *command* migration. Changed the fn name to post_command_migration.